packages/account/src/providers/provider.ts

```diff
@@ -645,6 +646,18 @@ export default class Provider {
    * @returns A promise that resolves to the estimated transaction request object.
    */
   async estimatePredicates(transactionRequest: TransactionRequest): Promise<TransactionRequest> {
+    const shouldEstimatePredicates = Boolean(
+      transactionRequest.inputs.find(
+        (input) =>
+          'predicate' in input &&
+          input.predicate &&
+          input.predicate !== getBytesCopy('0x') &&
+          new BN(input.predicateGasUsed).isZero()
```

Comment:
This check `new BN(input.predicateGasUsed).isZero()` allows for not rerunning predicate estimation because it `predicateGasUsed` is an indicator that the predicate has already been estimated.